drivers/xbee/xbee.c

```diff
@@ -493,10 +546,47 @@ int xbee_init(xbee_t *dev, uart_t uart, uint32_t baudrate,
     tmp[0] = (uint8_t)(XBEE_DEFAULT_PANID & 0xff);
     _set_panid(dev, tmp, 2);
 
+#if OPT_AES_ENCRYPTION
+    if (dev->encrypt_toggle){ /* Enable Encryption */
+        _set_aes_encryption(dev);
+    /* Write the key into the Xbee. */
+        _set_aes_encryption_key(dev,16);
+
+    }
+    else                     /* Disable Encryption */
+        _set_aes_encryption(dev);
+
+#endif
+
     DEBUG("xbee: Initialization successful\n");
     return 0;
 }
 
+int xbee_encrypt_config(xbee_t * dev, uint8_t * key_buf, unsigned int encryption_toggle)
```

Comment:
Rather than adding a function for this you can use `ng_netdev->driver->set()` for that by adding `NETCONF_OPT_ENCRYPTION` and `NETCONF_OPT_ENCRYPTION_KEY` to `ng_netconf.h`